### PR TITLE
Add REST API code from the chill-dashboard

### PR DESCRIPTION
**Pull request by badyakarprabin**
Extract the REST API express server code from [chill-dashboard](github.com/leapfrogtechnology/chill-dashboard) to this repository. 

**Note**: Some of the files here (eg: config.js, logger.js etc) are complete copy-paste from those from the chill repository, we'll need to extract these common codes, utils, and initialization/config stuff to a new package called `chill-core` to avoid code duplication.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (kabirbaidhya) [inline .eslintrc]: copy the `.eslintrc.yml` file from the chill repo instead. https://github.com/leapfrogtechnology/chill/blob/master/.eslintrc.yml

**Comment by badyakarprabin** [comment]
- Copied code from chill repo and pasted in chill rest api repo .

File: .nodemonignore [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Build directory
+dist
+
+# Test directory
+test
+
+# Other directories
+src/server/seeds
```
Comment (kabirbaidhya) [inline .nodemonignore]: src/server?

File: .postcssrc.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// https://github.com/michael-ciniawsky/postcss-load-config
```
Comment (kabirbaidhya) [inline .postcssrc.js]: This was a frontend specific file. Not needed here

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+language: node_js
```
Comment (kabirbaidhya) [inline .travis.yml]: Which editor are you using? It's putting unnecessary characters at the end.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  "version": "1.0.0",
+  "description": "Chill rest api - The monitoring tool.",
+  "scripts": {
+    "lint": "eslint src test; exit 0",
+    "lint:fix": "eslint src test --fix; exit 0",
+    "start:server": "nodemon --watch .env --watch src/ --exec babel-node src/ --source-maps"
+  },
+  "keywords": [
+    "chill",
+    "dashboard"
```
Comment (kabirbaidhya) [inline package.json]: rest-api

File: package.json [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    "babel-loader": "^6.2.10",
+    "babel-plugin-istanbul": "^4.1.1",
+    "babel-plugin-transform-es2015-modules-commonjs": "^6.24.0",
+    "babel-plugin-transform-react-remove-prop-types": "^0.3.3",
+    "babel-plugin-transform-runtime": "^6.15.0",
+    "babel-preset-env": "^1.3.2",
+    "babel-preset-es2015": "^6.18.0",
+    "babel-preset-es2016": "^6.16.0",
+    "babel-preset-es2017": "^6.16.0",
+    "babel-preset-react": "^6.23.0",
```
Comment (kabirbaidhya) [inline package.json]: remove this

File: package.json [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  "devDependencies": {
+    "autoprefixer": "^6.7.2",
+    "babel-cli": "^6.18.0",
+    "babel-core": "^6.22.1",
+    "babel-eslint": "^7.1.1",
+    "babel-jest": "^19.0.0",
+    "babel-loader": "^6.2.10",
+    "babel-plugin-istanbul": "^4.1.1",
+    "babel-plugin-transform-es2015-modules-commonjs": "^6.24.0",
+    "babel-plugin-transform-react-remove-prop-types": "^0.3.3",
```
Comment (kabirbaidhya) [inline package.json]: no need

File: package.json [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    "babel-plugin-transform-runtime": "^6.15.0",
+    "babel-preset-env": "^1.3.2",
+    "babel-preset-es2015": "^6.18.0",
+    "babel-preset-es2016": "^6.16.0",
+    "babel-preset-es2017": "^6.16.0",
+    "babel-preset-react": "^6.23.0",
+    "babel-register": "^6.18.0",
+    "chai": "^3.5.0",
+    "codecov": "^2.1.0",
+    "css-loader": "^0.28.0",
```
Comment (kabirbaidhya) [inline package.json]: this

File: package.json [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    "enzyme": "^2.7.1",
+    "eslint": "^3.19.0",
+    "eslint-plugin-react": "^6.10.0",
+    "file-loader": "^0.11.1",
```
Comment (kabirbaidhya) [inline package.json]: this

File: package.json [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+    "babel-preset-es2016": "^6.16.0",
+    "babel-preset-es2017": "^6.16.0",
+    "babel-preset-react": "^6.23.0",
+    "babel-register": "^6.18.0",
+    "chai": "^3.5.0",
+    "codecov": "^2.1.0",
+    "css-loader": "^0.28.0",
+    "enzyme": "^2.7.1",
+    "eslint": "^3.19.0",
+    "eslint-plugin-react": "^6.10.0",
```
Comment (kabirbaidhya) [inline package.json]: this

File: package.json [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    "file-loader": "^0.11.1",
+    "http-proxy-middleware": "^0.17.3",
+    "identity-obj-proxy": "^3.0.0",
+    "istanbul": "1.1.0-alpha.1",
+    "jest": "^19.0.2",
+    "mocha": "^3.2.0",
+    "node-sass": "^4.5.0",
```
Comment (kabirbaidhya) [inline package.json]: .

File: package.json [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+    "nodemon": "^1.11.0",
+    "postcss-loader": "^1.3.3",
```
Comment (kabirbaidhya) [inline package.json]: .

File: package.json [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+    "rimraf": "^2.5.4",
+    "sass-loader": "^6.0.2",
```
Comment (kabirbaidhya) [inline package.json]: .

File: package.json [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+    "semver": "^5.3.0",
+    "sinon": "^2.1.0",
+    "sinon-chai": "^2.8.0",
+    "supertest": "^2.0.1",
+    "url-loader": "^0.5.8"
```
Comment (kabirbaidhya) [inline package.json]: .

File: src/services/status.js [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+/**
+ * Get a service status.
+ *
+ * @param  {string|Number}  id
+ * @return {Promise}
+ */
+export async function getStatus(id) {
+  return await new Status({ id }).fetch()
```
Comment (kabirbaidhya) [inline src/services/status.js]:
```javascript
let status = await new Status({ id }).fetch();

if (!status) {
  throw new Boom.notFound('Service not found');
}

return status;
```

File: src/utils/logger.js [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+    new winston.transports.DailyRotateFile({
+      filename: `${logDir}/-debug.log`,
+      timestamp: tsFormat,
+      datePattern: 'yyyy-MM-dd',
+      prepend: true,
+      level: logLevel
+    })
+  ]
+});
+
```
Comment (kabirbaidhya) [inline src/utils/logger.js]: @badyakarprabin I think you've made some changes in this log format in the chill right? Copy that from here https://github.com/leapfrogtechnology/chill.

**Review by kabirbaidhya — DISMISSED** [review]

File: .travis.yml (hunk 4 above)
Comment (badyakarprabin) [reply .travis.yml]: VS code

File: .babelrc [hunk 17]
```diff
@@ -0,0 +1,16 @@
+{
+  "presets": [
+    ["env", { "modules": false }],
+    "es2015", "es2016", "es2017"
+  ],
+  "comments": false,
+  "env": {
```
Comment (kabirbaidhya) [inline .babelrc]: No need this block

File: .eslintrc (hunk 1 above)
Comment (kabirbaidhya) [inline .eslintrc]: Delete this file. We already have `.eslintrc.yml` file now.

**Comment by kabirbaidhya** [comment]
I'll finalize this PR.

**Review by mesaugat — APPROVED** [review]

**Comment by mesaugat** [comment]
The question is how many repositories will me make?

**Comment by kabirbaidhya** [comment]
Nice question :smile:.

**Comment by mesaugat** [comment]
And why do we need those configuration files here?

**Comment by kabirbaidhya** [comment]
They aren't actually separate config files, both the rest api and monitor needs to access the same `chill.yml` file. That's the issue. 
Even if we have `config.yml` for just monitor and `.env` file for the rest-api (as was before) we're going to have some config keys duplicated in both of the places and utils like logger are common for both. That's why we'll need to make the `chill.yml` file a centralized config file. 

Also, as we're thinking of moving the persistence logic from `chill` to the `rest-api`, it needs to access the `services` array of the `chill.yml` too.

**Pull request merged**